tests/unit/s2n_client_supported_versions_extension_test.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include <stdint.h>
+
+#include "tls/s2n_alerts.h"
+#include "tls/s2n_config.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_tls.h"
+#include "tls/extensions/s2n_client_supported_versions.h"
+
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_safety.h"
+
+#define PROTOCOL_VERSION_ALERT 70
+
+int get_alert(struct s2n_connection *conn) {
+    uint8_t error[2];
+    GUARD(s2n_stuffer_read_bytes(&conn->reader_alert_out, error, 2));
+    return error[1];
+}
+
+int write_test_supported_versions_list(struct s2n_stuffer *list, uint8_t *supported_versions, uint8_t length) {
+    GUARD(s2n_stuffer_write_uint8(list, length*S2N_TLS_PROTOCOL_VERSION_LEN));
```

Comment:
we usually have spaces around operators, here I helps avoid it looking like a pointer.